### PR TITLE
added rename function

### PR DESCRIPTION
**Pull request by NTyan**
Added rename function to editor

### CODE REVIEW

**Review by Antipkin-A — APPROVED** [review]

File: controllers/ApiController.php [hunk 1]
```diff
@@ -62,6 +62,43 @@ public function actionSaveas()
...
+            ]
+        ];
+
+        $response = $this->module->commandService($meta);
+
+        if($response['error'] !== 0){
+            throw new \Exception('Error from command Service: ' . $response['error']);
+        }
+
+        $file = File::findOne(['onlyoffice_key' => $data['key']]);
```
Comment (LinneyS) [inline controllers/ApiController.php:98]: change the order:
1) renaming
2) send command `meta`

File: controllers/ApiController.php [hunk 2]
```diff
@@ -62,6 +62,43 @@ public function actionSaveas()
...
+        $data = json_decode($body_data, TRUE);
+        if ($data === NULL) {
+            throw new \Exception('Could not parse json');
+        }
+
+        $meta = [
+            "c" => "meta",
+            "key" => $data['key'],
+            "meta"=> [
+                "title" => $data['newFileName']
```
Comment (LinneyS) [inline controllers/ApiController.php:88]: Send name with extension

File: resources/humhub.onlyoffice.js [hunk 3]
```diff
@@ -102,6 +102,7 @@ humhub.module('onlyoffice', function (module, require, $) {
         config.height = "100%";
         config.events = {
             'onRequestClose': onRequestClose,
+            'onMetaChange': onMetaChange,
```
Comment (LinneyS) [inline resources/humhub.onlyoffice.js:105]: Remove not used event

File: controllers/ApiController.php [hunk 4]
```diff
@@ -62,6 +62,43 @@ public function actionSaveas()
...
+
+        $response = $this->module->commandService($meta);
+
+        if($response['error'] !== 0){
+            throw new \Exception('Error from command Service: ' . $response['error']);
+        }
+
+        $file = File::findOne(['onlyoffice_key' => $data['key']]);
+        $ext = strtolower(FileHelper::getExtension($file));
+        $file->updateAttributes(['file_name' => $data['newFileName'] . "." . $ext]);
```
Comment (LinneyS) [inline controllers/ApiController.php:100]: Check access right

**Review by LinneyS — CHANGES_REQUESTED** [review]

**Pull request merged**